sql/opt/exec/explain: TestRetryFields failed

sql/opt/exec/explain.TestRetryFields [failed](https://tanzanite.cluster.engflow.com/invocations/default/69cbf539-3f56-48a3-b9d0-93515c22b27a?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4cGxhaW46ZXhwbGFpbl90ZXN0) on release-24.3 @ [fcc157ce6bd5eb383f69f04423b73e6dd80bf1d0](https://github.com/cockroachdb/cockroach/commits/fcc157ce6bd5eb383f69f04423b73e6dd80bf1d0):


```
=== RUN   TestRetryFields
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRetryFields2461801472
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    output_test.go:378: 
        	Error Trace:	pkg/sql/opt/exec/explain/output_test.go:378
        	Error:      	Should be true
        	Test:       	TestRetryFields
    output_test.go:379: -- test log scope end --
test logs left over in: outputs.zip/logTestRetryFields2461801472
--- FAIL: TestRetryFields (13.50s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetryFields.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48966

--- a/pkg/sql/opt/exec/explain/output_test.go
+++ b/pkg/sql/opt/exec/explain/output_test.go
@@ -360,22 +360,25 @@ func TestRetryFields(t *testing.T) {
 	sqlDB.Exec(t, "CREATE SEQUENCE s")
 
 	retryCountRE := regexp.MustCompile(`number of transaction retries: (\d+)`)
-	retryTimeRE := regexp.MustCompile(`time spent retrying the transaction: (\d+)[µsm]+`)
-	queryMatchRE := func(query string) bool {
-		rows, err := conn.QueryContext(ctx, query)
-		assert.NoError(t, err)
-		var foundCount, foundTime bool
-		for rows.Next() {
-			var res string
-			assert.NoError(t, rows.Scan(&res))
-			if matches := retryCountRE.FindStringSubmatch(res); len(matches) > 0 {
-				foundCount = true
-			}
-			if matches := retryTimeRE.FindStringSubmatch(res); len(matches) > 0 {
-				foundTime = true
-			}
+	retryTimeRE := regexp.MustCompile(`time spent retrying the transaction: ([\d\.]+)[µsm]+`)
+
+	const query = "EXPLAIN ANALYZE SELECT IF(nextval('s')<=3, crdb_internal.force_retry('1h'::INTERVAL), 0)"
+	rows, err := conn.QueryContext(ctx, query)
+	assert.NoError(t, err)
+	var output strings.Builder
+	var foundCount, foundTime bool
+	for rows.Next() {
+		var res string
+		assert.NoError(t, rows.Scan(&res))
+		output.WriteString(res)
+		output.WriteString("\n")
+		if matches := retryCountRE.FindStringSubmatch(res); len(matches) > 0 {
+			foundCount = true
 		}
-		return foundCount && foundTime
+		if matches := retryTimeRE.FindStringSubmatch(res); len(matches) > 0 {
+			foundTime = true
+		}
 	}
-	assert.True(t, queryMatchRE("EXPLAIN ANALYZE SELECT IF(nextval('s')<=3, crdb_internal.force_retry('1h'::INTERVAL), 0)"))
+	assert.Truef(t, foundCount, "expected to find transaction retries, full output:\n\n%s", output.String())
+	assert.Truef(t, foundTime, "expected to find time spent retrying, full output:\n\n%s", output.String())
 }

